### ISSUE TITLE
in BDD or Gherkin mode, the `After` is not synchronous where next scenario starts (while the `After` is still executing)

### ISSUE THREAD

**Issue by nnanoob** [Codeception/CodeceptJS#4288]
#### What are you trying to achieve?
In BDD / Gherkin mode, [After](https://codecept.io/bdd/#after) to be synchronous so it run to completion before the next scenario starts

Scenario 1
After --- run completely before Scenario 2 start
Scenario 2

#### What do you get instead?
`After` is asynchronous so it might still running while the next scenario already starting
It seems that the `After` implementation is based on [event](https://github.com/codeceptjs/CodeceptJS/blob/3.x/lib/container.js#L320-L322) which is independent from the promise chain.

### Details

* CodeceptJS version: 3.5.13
* NodeJS Version: v18.19.0
* Operating System: macOS
* webdriverio 8.32.0

**Comment by nnanoob** [comment]
this is closely related to this https://github.com/codeceptjs/CodeceptJS/issues/3870 but in this case, we are on `After` where we requires teardown of certain UI steps (which means we need to use the `I` and the approach of [recorder](https://codecept.io/hooks/#recorder) doesn't work for us) before proceeding to next scenario

**Comment by myrepojuly** [comment]
Agreed… this is really troubling a lot especially handling the after test events in the UI. Would be great if @kobenguyent you can fix this. I believe there was a similar issue with WebdriverIO and they fixed it.

**Comment by myrepojuly** [comment]
@DavertMik would you be able to help us with this issue? Pls let us of you need more info…

**Comment by DavertMik** [comment]
> It seems that the After implementation is based on [event](https://github.com/codeceptjs/CodeceptJS/blob/3.x/lib/container.js#L320-L322) which is independent from the promise chain.

@nnanoob @myrepojuly1987 

sorry, I don't think I can provide a quick fix on this

maybe you could use `recorder` to synchonize actions inside it?

```js
After(() => {
recorder.add(() => {
   // my actions
  })
})
```

If you have better suggestion, please send a PR

**Comment by myrepojuly** [comment]
@DavertMik Adding the actions to the recorder does not work and it just hangs. Sure, will check for better suggestions.

**Comment by myrepojuly** [comment]
@DavertMik @kobenguyent - could you please address this? At least if we could use the “I” object in the hooks, it would help us. Currently, adding the “I” object to the recorder inside a hook does nothing and just hangs.

**Comment by myrepojuly** [comment]
@nnanoob I totally forgot to add this comment earlier. While revisiting the issues I raised, I realized that as a workaround, you can always use the hooks available by extending the Helper ([https://codecept.io/helpers/#development](https://codecept.io/helpers/?utm_source=chatgpt.com#development)). It runs flawlessly without issues.

@kobenguyent  - just FYI
